internal/pkg/deploy/pipeline.go

```diff
@@ -637,6 +751,33 @@ func (stg *PipelineStage) buildDeploymentsGraph() *graph.Graph[string] {
 	return digraph
 }
 
+func (stg *PipelineStage) buildPrePostDeploymentsGraph(depType string) *graph.Graph[string] {
```

Comment:
Maybe we can reuse `buildDeploymentsGraph` if we pass either `stg.deployments`/`stg.preDeployments`/`stg.postDeployments`?